runner/src/scenario/ceramic/model_instance.rs

```diff
@@ -0,0 +1,717 @@
+use std::{str::FromStr, sync::Arc, time::Duration};
+
+use ceramic_http_client::{
+    api::{self, Pagination, StreamsResponse, StreamsResponseOrError},
+    ceramic_event::StreamId,
+    CeramicHttpClient, FilterQuery, ModelAccountRelation, ModelDefinition,
+};
+use goose::{metrics::GooseRequestMetric, prelude::*};
+use redis::AsyncCommands;
+use tracing::{debug, info, warn};
+
+use crate::{
+    goose_try,
+    scenario::{
+        ceramic::{
+            models::{self, RandomModelInstance},
+            CeramicClient, Credentials,
+        },
+        get_redis_client, is_goose_global_leader, is_goose_lead_user, is_goose_lead_worker,
+    },
+};
+
+use super::CeramicScenarioParameters;
+
+const SMALL_MODEL_ID_KEY: &str = "small_model_reuse_model_id";
+const LARGE_MODEL_ID_KEY: &str = "large_model_reuse_model_id";
+const SMALL_MID_ID_KEY: &str = "small_model_reuse_mid_id";
+const LARGE_MID_ID_KEY: &str = "large_model_reuse_mid_id";
+
+pub(crate) async fn set_key_to_stream_id(
+    conn: &mut redis::aio::Connection,
+    key: &str,
+    stream_id: &StreamId,
+) {
+    let _: () = conn.set(key, stream_id.to_string()).await.unwrap();
+}
+
+pub(crate) async fn loop_until_key_value_set(
+    conn: &mut redis::aio::Connection,
+    key: &str,
+) -> StreamId {
+    loop {
+        if conn.exists(key).await.unwrap() {
+            let id: String = conn.get(key).await.unwrap();
+            return StreamId::from_str(&id)
+                .map_err(|e| {
+                    tracing::error!("invalid stream: {:?} ", e);
+                    e
+                })
+                .unwrap();
+        } else {
+            tokio::time::sleep(Duration::from_millis(100)).await;
+        }
+    }
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct CountResponse {
```

Comment:
This struct (and the get counts query below) should probably be moved to the rust HTTP client.